File: cs/sdk/src/Proton.Drive.Sdk/Telemetry/UploadEvent.cs
using Proton.Sdk.Telemetry;

namespace Proton.Drive.Sdk.Telemetry;

public sealed class UploadEvent : IMetricEvent
{
    public string Name => "upload";

    public required VolumeType VolumeType { get; set; }

    public required long UploadedSize { get; set; }

    public required long ApproximateUploadedSize { get; set; }

    public required long ExpectedSize { get; set; }

    public UploadError? Error { get; set; }

    public string? OriginalError { get; set; }
}

File: kt/sdk/src/main/kotlin/me/proton/drive/sdk/telemetry/UploadEvent.kt
package me.proton.drive.sdk.telemetry

data class UploadEvent(
    val volumeType: VolumeType,
    val expectedSize: Long,
    val uploadedSize: Long,
    val approximateUploadedSize: Long,
    val error: UploadError? = null,
    val originalError: String? = null,
)
